src/query/storage/cache/redis_cache.go

```diff
@@ -41,3 +52,167 @@ func NewRedisCache(redisAddress string, logger *zap.Logger) *RedisCache {
 		logger:       logger,
 	}
 }
+
+// Given a fetch query, converts it into a key for Redis
+func keyEncode(query *storage.FetchQuery) string {
+	res := make([]string, len(query.TagMatchers))
+	for i, m := range query.TagMatchers {
+		res[i] = m.String()
+	}
+	// Sort to guarantee we have the same order every time
+	sort.Strings(res)
```

Comment:
Curious, why this need to be order?

Reply:
Just in case that two queries that ask for the same things but they come in a different order. For example, `http_requests{code=404, method="GET"}` and `http_requests{method="GET", code=404}` are the same but may not have the labels come in the same order so the key might look different without it